### PR TITLE
add support for steps(strides) in tensor slices

### PR DESCRIPTION
**Pull request by Krovatkin**

### CODE REVIEW

File: torch/csrc/jit/script/compiler.cpp [hunk 1]
```diff
@@ -2636,14 +2636,24 @@ struct to_ir {
       args.emplace_back(loc, "end", emitExpr(Expr(slice.end().get())));
     }
     if (input->type()->cast<TupleType>()) {
+
+      auto has_step = slice.step().present();
+      // TODO: add support for slicing tuples
+      if (has_step)
```
Comment (suo) [inline torch/csrc/jit/script/compiler.cpp]: nit: formatting

File: torch/csrc/jit/script/compiler.cpp [hunk 2]
```diff
@@ -2636,14 +2636,24 @@ struct to_ir {
...
+      {
+        throw ErrorReport(loc) << "Unsupported operation: slicing tuples isn't supported";
```
Comment (suo) [inline torch/csrc/jit/script/compiler.cpp]: I think this error message is not accurate, right? We do support tuple slices, just not steps

File: torch/csrc/jit/script/compiler.cpp [hunk 3]
```diff
@@ -2636,14 +2636,24 @@ struct to_ir {
...
+      }
+
       if (has_end) {
         return emitTupleSlice(loc, args[0], args[1], /*end*/ args[2]);
       } else {
         return emitTupleSlice(loc, args[0], args[1], c10::nullopt);
       }
     }
+
+    auto stride = emitExpr(Expr(slice.stepOr(1)));
```
Comment (suo) [inline torch/csrc/jit/script/compiler.cpp]: nit: "stride" is a heavyweight term in pytorch, let's avoid it and stick with step

**Review by suo — APPROVED** [review]
looks good! Sorry for the wait

File: torch/csrc/jit/script/compiler.cpp (hunk 2 above)
Comment (Krovatkin) [reply torch/csrc/jit/script/compiler.cpp]: will fix.

**Pull request closed without merging**